store/storetest/remote_cluster_store.go

```diff
@@ -0,0 +1,125 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+package storetest
+
+import (
+	"testing"
+
+	"github.com/mattermost/mattermost-server/v5/model"
+	"github.com/mattermost/mattermost-server/v5/store"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestRemoteClusterStore(t *testing.T, ss store.Store) {
+	t.Run("RemoteClusterSave", func(t *testing.T) { testRemoteClusterSave(t, ss) })
+	t.Run("RemoteClusterDelete", func(t *testing.T) { testRemoteClusterDelete(t, ss) })
+	t.Run("RemoteClusterGetAll", func(t *testing.T) { testRemoteClusterGetAll(t, ss) })
+}
+
+func testRemoteClusterSave(t *testing.T, ss store.Store) {
+
+	t.Run("Save", func(t *testing.T) {
+		rc := &model.RemoteCluster{
+			ClusterName: "some remote",
+			Hostname:    "somewhere.com",
+		}
+
+		rcSaved, err := ss.RemoteCluster().Save(rc)
+		require.Nil(t, err)
+		require.Equal(t, rcSaved.ClusterName, rc.ClusterName)
+		require.Equal(t, rcSaved.Hostname, rc.Hostname)
```

Comment:
I'd suggest inverting the parameters here since the first should be the *expected* value.